<Type Name="AccessViolationException" FullName="System.AccessViolationException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="eb360adfc3670886a06c5f3ab2daa2470318896a" />
    <Meta Name="ms.sourcegitcommit" Value="d31dc2ede16f6f7bc64e90d9f897ff54c4e3869b" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="es-ES" />
    <Meta Name="ms.lasthandoff" Value="04/03/2018" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class AccessViolationException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit AccessViolationException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.AccessViolationException" />
  <TypeSignature Language="VB.NET" Value="Public Class AccessViolationException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class AccessViolationException : SystemException" />
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <span data-ttu-id="a40ff-101">Excepción que se genera cuando se intenta leer o escribir en la memoria protegida.</span>
      <span class="sxs-lookup">
        <span data-stu-id="a40ff-101">The exception that is thrown when there is an attempt to read or write protected memory.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="a40ff-102">Se produce una infracción de acceso en código no administrado o no seguro cuando el código intenta leer o escribir en memoria que no se ha asignado o a que no tiene acceso.</span><span class="sxs-lookup"><span data-stu-id="a40ff-102">An access violation occurs in unmanaged or unsafe code when the code attempts to read or write to memory that has not been allocated, or to which it does not have access.</span></span> <span data-ttu-id="a40ff-103">Esto suele ocurrir porque un puntero tiene un valor incorrecto.</span><span class="sxs-lookup"><span data-stu-id="a40ff-103">This usually occurs because a pointer has a bad value.</span></span> <span data-ttu-id="a40ff-104">No todas las lecturas o escrituras con punteros no válidos llevan a infracciones de acceso; por lo tanto, una infracción de acceso suele indicar que se han realizado varias lecturas o escrituras con punteros no válidos y se ha dañado la memoria.</span><span class="sxs-lookup"><span data-stu-id="a40ff-104">Not all reads or writes through bad pointers lead to access violations, so an access violation usually indicates that several reads or writes have occurred through bad pointers, and that memory might be corrupted.</span></span> <span data-ttu-id="a40ff-105">Por lo tanto, las infracciones de acceso casi siempre indican errores de programación graves.</span><span class="sxs-lookup"><span data-stu-id="a40ff-105">Thus, access violations almost always indicate serious programming errors.</span></span> <span data-ttu-id="a40ff-106">Un <xref:System.AccessViolationException> identifique claramente estos errores graves.</span><span class="sxs-lookup"><span data-stu-id="a40ff-106">An <xref:System.AccessViolationException> clearly identifies these serious errors.</span></span>  
  
 <span data-ttu-id="a40ff-107">En los programas que se compone únicamente de código administrado comprobable, todas las referencias son válidas o nulas y son imposibles de infracciones de acceso.</span><span class="sxs-lookup"><span data-stu-id="a40ff-107">In programs consisting entirely of verifiable managed code, all references are either valid or null, and access violations are impossible.</span></span> <span data-ttu-id="a40ff-108">Cualquier operación que intenta hacer referencia a una referencia nula en código comprobable produce una <xref:System.NullReferenceException> excepción.</span><span class="sxs-lookup"><span data-stu-id="a40ff-108">Any operation that attempts to reference a null reference in verifiable code throws a <xref:System.NullReferenceException> exception.</span></span> <span data-ttu-id="a40ff-109">Un <xref:System.AccessViolationException> se produce solo cuando el código administrado comprobable interactúa con código no administrado o con código administrado no seguro.</span><span class="sxs-lookup"><span data-stu-id="a40ff-109">An <xref:System.AccessViolationException> occurs only when verifiable managed code interacts with unmanaged code or with unsafe managed code.</span></span>  
  
## <a name="troubleshooting-accessviolationexception-exceptions"></a><span data-ttu-id="a40ff-110">Solucionar problemas de excepciones AccessViolationException</span><span class="sxs-lookup"><span data-stu-id="a40ff-110">Troubleshooting AccessViolationException exceptions</span></span>  
 <span data-ttu-id="a40ff-111">Un <xref:System.AccessViolationException> excepción solo se produce en código administrado no seguro o cuando administra comprobable código interactúa con código no administrado:</span><span class="sxs-lookup"><span data-stu-id="a40ff-111">An <xref:System.AccessViolationException> exception can occur only in unsafe managed code or when verifiable managed code interacts with unmanaged code:</span></span>  
  
-   <span data-ttu-id="a40ff-112">Una infracción de acceso que se produce en código administrado no seguro se puede expresar como un <xref:System.NullReferenceException> excepción o un <xref:System.AccessViolationException> excepción, dependiendo de la plataforma.</span><span class="sxs-lookup"><span data-stu-id="a40ff-112">An access violation that occurs in unsafe managed code can be expressed as either a <xref:System.NullReferenceException> exception or an <xref:System.AccessViolationException> exception, depending on the platform.</span></span>  
  
-   <span data-ttu-id="a40ff-113">Una infracción de acceso en código no administrado que se propaga hasta el código administrado siempre se ajusta en un <xref:System.AccessViolationException> excepción.</span><span class="sxs-lookup"><span data-stu-id="a40ff-113">An access violation in unmanaged code that bubbles up to managed code is always wrapped in an <xref:System.AccessViolationException> exception.</span></span>  
  
 <span data-ttu-id="a40ff-114">En cualquier caso, puede identificar y corregir la causa de la <xref:System.AccessViolationException> excepción como sigue:</span><span class="sxs-lookup"><span data-stu-id="a40ff-114">In either case, you can identify and correct the cause of the <xref:System.AccessViolationException> exception as follows:</span></span>  
  
 <span data-ttu-id="a40ff-115">Asegúrese de que la memoria a la que intenta obtener acceso se haya asignado.</span><span class="sxs-lookup"><span data-stu-id="a40ff-115">Make sure that the memory that you are attempting to access has been allocated.</span></span>  
 <span data-ttu-id="a40ff-116">Un <xref:System.AccessViolationException> excepción siempre se produce un intento para tener acceso a la memoria protegida, es decir, para tener acceso a memoria que no está asignada o que no pertenece a un proceso.</span><span class="sxs-lookup"><span data-stu-id="a40ff-116">An <xref:System.AccessViolationException> exception is always thrown by an attempt to access protected memory -- that is, to access memory that is not allocated or that is not owned by a process.</span></span>  
  
 <span data-ttu-id="a40ff-117">La administración automática de la memoria es uno de los servicios que proporciona Common Language Runtime.</span><span class="sxs-lookup"><span data-stu-id="a40ff-117">Automatic memory management is one of the services that the common language runtime provides.</span></span> <span data-ttu-id="a40ff-118">Si el código administrado proporciona la misma funcionalidad que el código unmanagede, puede desplazarse al código administrado para aprovechar esta funcionalidad.</span><span class="sxs-lookup"><span data-stu-id="a40ff-118">If managed code provides the same functionality as your unmanagede code, you may wish to move to managed code to take advantage of this functionality.</span></span> <span data-ttu-id="a40ff-119">Para obtener más información, consulte [la administración automática de memoria](~/docs/standard/automatic-memory-management.md).</span><span class="sxs-lookup"><span data-stu-id="a40ff-119">For more information, see [Automatic Memory Management](~/docs/standard/automatic-memory-management.md).</span></span>  
  
 <span data-ttu-id="a40ff-120">Asegúrese de que la memoria a la que intenta obtener acceso no esté dañada.</span><span class="sxs-lookup"><span data-stu-id="a40ff-120">Make sure that the memory that you are attempting to access has not been corrupted.</span></span>  
 <span data-ttu-id="a40ff-121">Si se han realizado varias operaciones de lectura o escritura con punteros no válidos, es posible que la memoria esté dañada.</span><span class="sxs-lookup"><span data-stu-id="a40ff-121">If several read or write operations have occurred through bad pointers, memory may be corrupted.</span></span> <span data-ttu-id="a40ff-122">Esto suele ocurrir al leer o escribir en direcciones fuera de un búfer predefinido.</span><span class="sxs-lookup"><span data-stu-id="a40ff-122">This typically occurs when reading or writing to addresses outside of a predefined buffer.</span></span>  
  
## <a name="accessviolationexception-and-trycatch-blocks"></a><span data-ttu-id="a40ff-123">Bloques try/catch y de AccessViolationException</span><span class="sxs-lookup"><span data-stu-id="a40ff-123">AccessViolationException and try/catch blocks</span></span>  
 <span data-ttu-id="a40ff-124">A partir de la [!INCLUDE[net_v40_long](~/includes/net-v40-long-md.md)], <xref:System.AccessViolationException> las excepciones producidas por common language runtime no son controladas por el `catch` instrucción en un controlador de excepciones estructurado si la excepción se produce fuera de la memoria reservada mediante common language runtime.</span><span class="sxs-lookup"><span data-stu-id="a40ff-124">Starting with the [!INCLUDE[net_v40_long](~/includes/net-v40-long-md.md)], <xref:System.AccessViolationException> exceptions thrown by the common language runtime are not handled by the `catch` statement in a structured exception handler if the exception occurs outside of the memory reserved by the common language runtime.</span></span> <span data-ttu-id="a40ff-125">Para administrar como un <xref:System.AccessViolationException> excepción, debe aplicar el <xref:System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptionsAttribute> atributo al método en el que se produce la excepción.</span><span class="sxs-lookup"><span data-stu-id="a40ff-125">To handle such an <xref:System.AccessViolationException> exception, you should apply the  <xref:System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptionsAttribute> attribute to the method in which the exception is thrown.</span></span> <span data-ttu-id="a40ff-126">Este cambio no afecta a <xref:System.AccessViolationException> las excepciones producidas por código de usuario, lo que puede seguir para capturar un `catch` instrucción.</span><span class="sxs-lookup"><span data-stu-id="a40ff-126">This change does not affect <xref:System.AccessViolationException> exceptions thrown by user code, which can continue to be caught by a `catch` statement.</span></span> <span data-ttu-id="a40ff-127">Para el código escrito para versiones anteriores de .NET Framework que desea volver a compilar y ejecutar sin modificaciones el [!INCLUDE[net_v40_long](~/includes/net-v40-long-md.md)], puede agregar el [ \<legacyCorruptedStateExceptionsPolicy >](~/docs/framework/configure-apps/file-schema/runtime/legacycorruptedstateexceptionspolicy-element.md) elemento al archivo de configuración de la aplicación.</span><span class="sxs-lookup"><span data-stu-id="a40ff-127">For code written for previous versions of the .NET Framework that you want to recompile and run without modification on the [!INCLUDE[net_v40_long](~/includes/net-v40-long-md.md)], you can add the [\<legacyCorruptedStateExceptionsPolicy>](~/docs/framework/configure-apps/file-schema/runtime/legacycorruptedstateexceptionspolicy-element.md) element to your app's configuration file.</span></span> <span data-ttu-id="a40ff-128">Tenga en cuenta que también puede recibir notificación de las excepciones si se ha definido un controlador para el <xref:System.AppDomain.FirstChanceException?displayProperty=nameWithType> o <xref:System.AppDomain.UnhandledException?displayProperty=nameWithType> eventos.</span><span class="sxs-lookup"><span data-stu-id="a40ff-128">Note that you can also receive notification of the exceptions if you have defined a handler for the <xref:System.AppDomain.FirstChanceException?displayProperty=nameWithType> or <xref:System.AppDomain.UnhandledException?displayProperty=nameWithType> event.</span></span>  
  
## <a name="version-information"></a><span data-ttu-id="a40ff-129">Información de versión</span><span class="sxs-lookup"><span data-stu-id="a40ff-129">Version Information</span></span>  
 <span data-ttu-id="a40ff-130">Esta excepción se produce en el .NET Framework 2.0 y versiones posteriores.</span><span class="sxs-lookup"><span data-stu-id="a40ff-130">This exception is thrown in the .NET Framework 2.0 and later versions.</span></span> <span data-ttu-id="a40ff-131">En versiones anteriores de .NET Framework, una infracción de acceso en código no administrado o código administrado no seguro se representa mediante un <xref:System.NullReferenceException> en código administrado.</span><span class="sxs-lookup"><span data-stu-id="a40ff-131">In earlier versions of the .NET Framework, an access violation in unmanaged code or unsafe managed code is represented by a <xref:System.NullReferenceException> in managed code.</span></span> <span data-ttu-id="a40ff-132">Un <xref:System.NullReferenceException> también se produce cuando se desreferencia una referencia nula en código administrado comprobable, una aparición que implican datos dañados, y no hay ninguna manera de distinguir entre estas dos situaciones en las versiones 1.0 o 1.1.</span><span class="sxs-lookup"><span data-stu-id="a40ff-132">A <xref:System.NullReferenceException> is also thrown when a null reference is dereferenced in verifiable managed code, an occurrence that does not involve data corruption, and there is no way to distinguish between the two situations in versions 1.0 or 1.1.</span></span>  
  
 <span data-ttu-id="a40ff-133">Los administradores pueden permitir que las aplicaciones seleccionadas revertir al comportamiento de la versión 1.1 de .NET Framework.</span><span class="sxs-lookup"><span data-stu-id="a40ff-133">Administrators can allow selected applications to revert to the behavior of the .NET Framework version 1.1.</span></span> <span data-ttu-id="a40ff-134">Coloque la siguiente línea en el [ &lt;en tiempo de ejecución&gt; elemento](~/docs/framework/configure-apps/file-schema/runtime/runtime-element.md) sección del archivo de configuración de la aplicación:</span><span class="sxs-lookup"><span data-stu-id="a40ff-134">Place the following line in the [&lt;runtime&gt; Element](~/docs/framework/configure-apps/file-schema/runtime/runtime-element.md) section of the configuration file for the application:</span></span>  
  
```  
<legacyNullReferenceExceptionPolicy enabled = "1"/>  
```  
  
 ]]></format>
    </remarks>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="a40ff-135">Inicializa una nueva instancia de la clase <see cref="T:System.AccessViolationException" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-135">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AccessViolationException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AccessViolationException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AccessViolationException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="a40ff-136">Inicializa una nueva instancia de la clase <see cref="T:System.AccessViolationException" /> con un mensaje proporcionado por el sistema que describe el error.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-136">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class with a system-supplied message that describes the error.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="a40ff-137">Este constructor inicializa la <xref:System.Exception.Message%2A> propiedad de la nueva instancia en un mensaje proporcionado por el sistema que describe el error, como "intento de leer o escribir en la memoria protegida.</span><span class="sxs-lookup"><span data-stu-id="a40ff-137">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "Attempted to read or write protected memory.</span></span> <span data-ttu-id="a40ff-138">Esto suele ser un valor que indica que otra memoria está dañada."</span><span class="sxs-lookup"><span data-stu-id="a40ff-138">This is often an indication that other memory has been corrupted."</span></span> <span data-ttu-id="a40ff-139">Este mensaje tiene en cuenta la referencia cultural del sistema actual.</span><span class="sxs-lookup"><span data-stu-id="a40ff-139">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="a40ff-140">En la tabla siguiente se muestran los valores de propiedad iniciales para una instancia de <xref:System.AccessViolationException>.</span><span class="sxs-lookup"><span data-stu-id="a40ff-140">The following table shows the initial property values for an instance of <xref:System.AccessViolationException>.</span></span>  
  
|<span data-ttu-id="a40ff-141">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="a40ff-141">Property</span></span>|<span data-ttu-id="a40ff-142">Valor</span><span class="sxs-lookup"><span data-stu-id="a40ff-142">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="a40ff-143">`null`.</span><span class="sxs-lookup"><span data-stu-id="a40ff-143">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="a40ff-144">La cadena del mensaje de error localizado.</span><span class="sxs-lookup"><span data-stu-id="a40ff-144">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AccessViolationException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AccessViolationException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AccessViolationException(System::String ^ message);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="a40ff-145">Mensaje que describe la excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-145">The message that describes the exception.</span>
          </span>
          <span data-ttu-id="a40ff-146">El autor de llamada de este constructor debe asegurarse de que la cadena se haya adaptado para la referencia cultural actual del sistema.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-146">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="a40ff-147">Inicializa una nueva instancia de la clase <see cref="T:System.AccessViolationException" /> con un mensaje especificado que describe el error.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-147">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class with a specified message that describes the error.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="a40ff-148">En la tabla siguiente se muestran los valores de propiedad iniciales para una instancia de <xref:System.AccessViolationException>.</span><span class="sxs-lookup"><span data-stu-id="a40ff-148">The following table shows the initial property values for an instance of <xref:System.AccessViolationException>.</span></span>  
  
|<span data-ttu-id="a40ff-149">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="a40ff-149">Property</span></span>|<span data-ttu-id="a40ff-150">Valor</span><span class="sxs-lookup"><span data-stu-id="a40ff-150">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="a40ff-151">`null`.</span><span class="sxs-lookup"><span data-stu-id="a40ff-151">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="a40ff-152">La cadena de mensaje de error especificada en `message`.</span><span class="sxs-lookup"><span data-stu-id="a40ff-152">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected AccessViolationException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AccessViolationException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; AccessViolationException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="a40ff-153">
            <see cref="T:System.Runtime.Serialization.SerializationInfo" /> que contiene los datos del objeto serializados.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-153">The <see cref="T:System.Runtime.Serialization.SerializationInfo" /> that holds the serialized object data.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="a40ff-154">
            <see cref="T:System.Runtime.Serialization.StreamingContext" /> que contiene información contextual sobre el origen o el destino.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-154">The <see cref="T:System.Runtime.Serialization.StreamingContext" /> that contains contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="a40ff-155">Inicializa una nueva instancia de la clase <see cref="T:System.AccessViolationException" /> con datos serializados.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-155">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class with serialized data.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="a40ff-156">Durante la deserialización se llama a este constructor para reconstituir el objeto de excepción transmitido en una secuencia.</span><span class="sxs-lookup"><span data-stu-id="a40ff-156">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="a40ff-157">Para obtener más información, consulte [serialización XML y SOAP](~/docs/standard/serialization/xml-and-soap-serialization.md).</span><span class="sxs-lookup"><span data-stu-id="a40ff-157">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public AccessViolationException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.AccessViolationException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; AccessViolationException(System::String ^ message, Exception ^ innerException);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="a40ff-158">Mensaje que describe la excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-158">The message that describes the exception.</span>
          </span>
          <span data-ttu-id="a40ff-159">El autor de llamada de este constructor debe asegurarse de que la cadena se haya adaptado para la referencia cultural actual del sistema.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-159">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="a40ff-160">Excepción que es la causa de la excepción actual.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-160">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="a40ff-161">Si el parámetro <c>innerException</c> no es <see langword="null" />, se genera la excepción actual en un bloque <see langword="catch" /> que controla la excepción interna.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-161">If the <c>innerException</c> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="a40ff-162">Inicializa una nueva instancia de la clase <see cref="T:System.AccessViolationException" /> con el mensaje de error especificado y una referencia a la excepción interna que representa la causa de esta excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="a40ff-162">Initializes a new instance of the <see cref="T:System.AccessViolationException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="a40ff-163">Una excepción que se produce como consecuencia directa de una excepción anterior debe incluir una referencia a esta última en la propiedad <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="a40ff-163">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="a40ff-164">La propiedad <xref:System.Exception.InnerException%2A> devuelve el mismo valor que se haya pasado al constructor, o bien `null` en caso de que la propiedad <xref:System.Exception.InnerException%2A> no le haya proporcionado al constructor el valor de la excepción interna.</span><span class="sxs-lookup"><span data-stu-id="a40ff-164">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="a40ff-165">En la tabla siguiente se muestran los valores de propiedad iniciales para una instancia de <xref:System.AccessViolationException>.</span><span class="sxs-lookup"><span data-stu-id="a40ff-165">The following table shows the initial property values for an instance of <xref:System.AccessViolationException>.</span></span>  
  
|<span data-ttu-id="a40ff-166">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="a40ff-166">Property</span></span>|<span data-ttu-id="a40ff-167">Valor</span><span class="sxs-lookup"><span data-stu-id="a40ff-167">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="a40ff-168">`null`.</span><span class="sxs-lookup"><span data-stu-id="a40ff-168">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="a40ff-169">La cadena de mensaje de error especificada en `message`.</span><span class="sxs-lookup"><span data-stu-id="a40ff-169">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>